### ISSUE TITLE
add architecture specific dependency fields support

### ISSUE THREAD

**Issue by rmarquis** [rmarquis/pacaur#465]
Currently, the RPC interface doesn't support architecture specific fields. It merges them in the generic arrays instead. See [FS#48796](https://bugs.archlinux.org/task/48796).

Partial support for arch specific dependencies has been implemented with a temporary workaround in 6e3cb3b933b2f37b67a8173ff5d461ad8058f12c. This will however work only if the 32 bits (make)dependencies use the naming prefix `lib32-` on on the x86_64 arch, and the regular name on the i686 arch. Most user cases should be covered by this hack, but probably not all (I'm still looking for a test case where this hack actually fails).

It is possible to implement another temporary hack by reading and parsing the .SRCINFO files to retrieve the correct information. This indeed bypasses the RPC interface, and is [implemented by bauerbill](https://bbs.archlinux.org/viewtopic.php?pid=1617103#p1617103) and pkgbuilder. This would however require to substantially change the current design as PKGBUILD cloning is done after dependency resolution, and this is something I'm not going to do. I will instead wait for the proper fix to be implemented in the RPC. Once this is done, correct arch specific support will be trivial to add and the existing hack can be removed.

See also [wiki discussion](https://wiki.archlinux.org/index.php/Talk:AUR_helpers) and https://github.com/AladW/aurutils/issues/80.

**Comment by Lucki** [comment]
> I'm still looking for a test case where this hack actually fails

I hope this is the same issue:
Before building openrct2-git on a 32-bit-machine, pacaur fails at `resolving dependencies…` with `no results found for gcc-multilib` which isn't available for 32-bit-systems.
The dependency is declared for 64-bit `makedepends_x86_64=('gcc-multilib')` so it should affect 32-bit.

**Comment by rmarquis** [comment]
Yes, this is the same issue and a situation where the simplistic hack above fails. Unfortunately, handling this package requires [FS#48796](https://bugs.archlinux.org/task/48796) to be fixed first as explained above.

**Comment by rmarquis** [comment]
I've expanded the current hack to handle gcc-multilib as it's always needed with lib32 packages on 64 bit arch anyway (see 4d36c53779d9a83480e351ba62589cc4d11aa6dc).

**Comment by rmarquis** [comment]
Note: Arch is [phasing out i686 support](https://www.archlinux.org/news/phasing-out-i686-support/). No further work is therefore needed, and the current hack could be removed in November 2017.

**Comment by ArchangeGabriel** [comment]
@rmarquis You got it wrong I think. Official i686 support in the repo from current devs is what is going out.

i686 will probably stay, the ideal goal being to have it built in an automated manner by the end of the year. But what is sure, is that support for multiple architectures will stay in all the tools.

That being said, I’m not asking you to spend any of your time on this issue until at least the future of i686 is established. ;)

**Comment by rmarquis** [comment]
True, the *official* support is ending. We'll see if a viable community effort happens.

**Comment by rmarquis** [comment]
This is superseded by #720, closing now.
